src/components/contextual/pages/pool/PoolCompositionCard/PoolCompositionCard.vue

```diff
@@ -0,0 +1,160 @@
+<script setup lang="ts">
+import useBreakpoints from '@/composables/useBreakpoints';
+import { bnum } from '@/lib/utils';
+import {
+  getUnderlyingTokens,
+  findTokenByAddress,
+} from './nested-token-helpers';
+import { Pool } from '@/services/pool/types';
+import useWeb3 from '@/services/web3/useWeb3';
+import { toRefs } from 'vue';
+
+import AssetRow from './components/AssetRow.vue';
+import useNumbers, { FNumFormats } from '@/composables/useNumbers';
+import useTokens from '@/composables/useTokens';
+import { isWeightedLike } from '@/composables/usePool';
+
+/**
+ * TYPES
+ */
+type Props = {
+  pool: Pool;
+};
+
+/**
+ * PROPS
+ */
+const props = defineProps<Props>();
+const { pool } = toRefs(props);
+const isWeighted = isWeightedLike(pool.value.poolType);
+
+/**
+ * COMPOSABLES
+ */
+const { upToLargeBreakpoint } = useBreakpoints();
+const { explorerLinks } = useWeb3();
+const { fNum2 } = useNumbers();
+const { priceFor } = useTokens();
+
+/**
+ * METHODS
+ */
+function getTokenShare(address: string) {
+  const linearPools = pool.value.onchain?.linearPools;
+
+  if (linearPools == null) {
+    return null;
+  }
+
+  const token = props.pool?.onchain?.tokens[address];
+
+  return bnum(token?.balance || '0')
+    .div(linearPools[address].totalSupply)
+    .toString();
+}
+
+function hasNestedTokens(address: string) {
+  return getUnderlyingTokens(pool.value, address).length;
+}
+
+function getRootTokenStyle(address) {
+  // When there are nested tokens, the root token will have full width (just breakdown)
+  if (hasNestedTokens(address)) return 'py-4';
+  // When NO nested tokens, the root token will have 3 cols (breakdown + balance + value)
+  if (isWeighted) return 'grid grid-cols-4 py-4';
```

Comment:
There are many other pool types that don't have nesting. This should be checking if it's a nested pool rather than for a specific pool type, e.g.
```suggestion
  if (!isDeep(pool)) return 'grid grid-cols-4 py-4';
```

Reply:
I don't understand the suggestion. I was considering 3 cases: 
1. hasNestedTokens() like **Balancer Boosted Aave USD**
<img width="827" alt="nestedPool" src="https://user-images.githubusercontent.com/1316240/200001031-f83ef793-5b34-47e5-8bce-1082a19e39f7.png">
2. isWeighted which I understood (probably wrongly) as pools which have weight atribute like **Oracle Weighted Pool**:
<img width="790" alt="weighted" src="https://user-images.githubusercontent.com/1316240/200001673-93f9b87c-2acf-4c30-afbd-237f946af2bb.png">
3. The rest of the pools (no nested, no weighted) like this **stable pool**:
<img width="804" alt="Screenshot 2022-11-04 at 15 38 06" src="https://user-images.githubusercontent.com/1316240/200001926-f1e3cc68-3f9f-4aec-95a2-691e69dc729d.png">

I still miss a lot of domain background 😅 Maybe we can have a quick call to clarify this?

Reply:
ahh my bad, was skim-reading this part and didn't fully understand it. You are right here.

Reply:
Maybe my new helper: 
```ts
function hasNestedTokens(address: string) {
  return getUnderlyingTokens(pool.value, address).length;
}
```
can be avoided and replaced by `isDeep`?